src/System.Text.Json/tests/Utf8JsonReaderTests.TryGet.cs

```diff
@@ -263,9 +290,202 @@ public static void TestingNumbersInvalidConversionToInt64(string jsonString, dou
+                {
+                    #region TryGet calls
+                    Assert.False(json.TryGetUInt32(out uint value));
+                    Assert.True(json.TryGetDouble(out double doubleValue));
+                    Assert.Equal(expected, doubleValue);
+                    #endregion
+
+                    #region Get calls
+                    try
+                    {
+                        json.GetUInt32();
+                        Assert.True(false, "Expected GetUInt32 to throw FormatException.");
+                    }
+                    catch (FormatException)
+                    {
+                        /* Expected exception */
+                    }
+                    doubleValue = json.GetDouble();
+                    Assert.Equal(expected, doubleValue);
+                    #endregion
+                }
+            }
+
+            Assert.Equal(dataUtf8.Length, json.BytesConsumed);
+            Assert.Equal(json.BytesConsumed, json.CurrentState.BytesConsumed);
+        }
+
+        [Theory]
+        [InlineData("0.000", 0)]
+        [InlineData("1e1", 10)]
+        [InlineData("1.1e2", 110)]
+        [InlineData("12345.1", 12345.1)]
+        [InlineData("123456789012345678901", 123456789012345678901d)]
+        [InlineData("-1", -1)] // ulong.MinValue - 1
+        public static void TestingNumbersInvalidConversionToUInt64(string jsonString, double expected)
+        {
+            byte[] dataUtf8 = Encoding.UTF8.GetBytes(jsonString);
+
+            var json = new Utf8JsonReader(dataUtf8, true, default);
+            while (json.Read())
+            {
+                if (json.TokenType == JsonTokenType.Number)
+                {
+                    #region TryGet calls
+                    Assert.False(json.TryGetUInt64(out ulong value));
+                    Assert.True(json.TryGetDouble(out double doubleValue));
+                    Assert.Equal(expected, doubleValue);
+                    #endregion
+
+                    #region Get calls
+                    try
+                    {
+                        json.GetUInt64();
+                        Assert.True(false, "Expected GetInt64 to throw FormatException.");
+                    }
+                    catch (FormatException)
+                    {
+                        /* Expected exception */
+                    }
+                    doubleValue = json.GetDouble();
+                    Assert.Equal(expected, doubleValue);
+                    #endregion
+                }
+            }
+
+            Assert.Equal(dataUtf8.Length, json.BytesConsumed);
+            Assert.Equal(json.BytesConsumed, json.CurrentState.BytesConsumed);
+        }
+
+        [Theory]
+        [InlineData("-4.402823E+38", float.NegativeInfinity)] // float.MinValue - 1
+        [InlineData("4.402823E+38", float.PositiveInfinity)]  // float.MaxValue + 1
+        public static void TestingNumbersInvalidConversionToSingle(string jsonString, float expected)
```

Comment:
Given we are fine with handling really large numbers and return True with some infinity value, we should change the test name (it isn't invalid).